File: src/main/java/com/example/ssauc/user/product/repository/CategoryRepository.java
package com.example.ssauc.user.product.repository;

import com.example.ssauc.user.product.entity.Category;
import com.example.ssauc.user.product.entity.Product;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface CategoryRepository extends JpaRepository<Category, Long> {

    Optional<Category> findByName(String name);
    List<Category> findAll();
}

File: src/main/java/com/example/ssauc/user/product/repository/CategoryRepository.kt
package com.example.ssauc.user.product.repository

import com.example.ssauc.user.product.entity.Category
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface CategoryRepository : JpaRepository<Category?, Long?> {
    fun findByName(name: String?): Optional<Category?>
    override fun findAll(): List<Category?>
}